clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -760,14 +765,19 @@ public void commitAsync(Map<TopicPartition, OffsetAndMetadata> offsets, OffsetCo
             // waiting for a response.
             CompletableFuture<Void> future = commit(offsets, false, Optional.empty());
             future.whenComplete((r, t) -> {
+
+                if (t == null) {
+                    offsetCommitCallbackInvoker.submitCommitInterceptors(offsets);
+                }
+
                 if (callback == null) {
                     if (t != null) {
                         log.error("Offset commit with offsets {} failed", offsets, t);
                     }
                     return;
                 }
 
-                invoker.submit(new OffsetCommitCallbackTask(callback, offsets, (Exception) t));
+                offsetCommitCallbackInvoker.submitUserCallback(callback, offsets, (Exception) t);
```

Comment:
Refactored so that `OffsetCommitCallbackTask` can be an private class inside the invoker.